### PR TITLE
Convert operations(operator, motion, text-object) to JS

### PR DESCRIPTION
**Pull request by t9md**
To let me understand how it is, clarify requirement.

# TODO( 🚢  released)

- [x] Eliminate 2nd param of Base `constructor` and if necessary, modify props **after** construction.
- [x] Update example code in [Wiki](https://github.com/t9md/atom-vim-mode-plus/wiki/ExtendVimModePlusInInitFile)
- [ ] ~~Find the way to notify user to inform extending JS class by coffee is no longer supported.~~ I don't thins there is it.
  - Refs: https://github.com/jashkenas/coffeescript/issues/4233
- [ ] Convert coffee based vmp-plugin to JS.
  - [x] vmp-ex-mode t9md/atom-vim-mode-plus-ex-mode#27 🚢
  - [x] vmp-jasmine-increase-focus t9md/atom-vim-mode-plus-jasmine-increase-focus#4 🚢
  - [x] vmp-move-to-symbols t9md/atom-vim-mode-plus-move-to-symbols#2 🚢
  - [x] vmp-move-selected-text t9md/atom-vim-mode-plus-move-selected-text#6 🚢
  - [x] vmp-replace-with-execution t9md/atom-vim-mode-plus-replace-with-execution#1 🚢
  - [x] vmp-plugin-sample t9md/atom-vim-mode-plus-plugin-sample#1 🚢
  - [x] quick-highlight t9md/atom-quick-highlight#24 🚢
  - [x] vim-mode-plus-exchange dillonkearns/atom-vim-mode-plus-exchange#3
  - [x] vim-mode-plus-duplicate dillonkearns/atom-vim-mode-plus-duplicate#1
  - [ ] artarf/git-diff-staged artarf/git-diff-staged#1

# Migration best practices memo

- Basic rule is to move logic to from `constructor` or `initialize` to `execute()`.
- In ES6 class's `constructor`, `super` must be called first to refer `this`.
- So we no longer do following logic.
  - Old way in CoffeeScript
    1. set `this.prop`
    2. then call `super`
    3. In parent's `constructor` behave differently based on `this.prop`
  - New way in ES6
    1. always call `super` first in `constructor`
    2. set `this.prop`
    3. in `execute`, behave differently based on `this.prop`

# Working memo

### before conversion

had 2 layer

- klass.prototype.ivar
- instance.ivar

### after conversion

only 1 layer (no `prototype.ivar` layer)

- instance.ivar

# Caution

- ES6 class is not extendable by CS's class.
- Thus, converting breaks user's custom class in their `init.coffee`(if exists) or vmp-plugin package written in coffee.
- CS's executable class body is no longer availale.
- Class prop is not set on `prototype`, and it's set **after** constructor's `super` call.
- So don't expect these class props are available in parent's constructor.
  - Basically keep `constructor` function small and simple.
- In CS, `Object.assign(this, prop)` in constructor can overwrite properties defined in prototype, but  in JS, properties are set **after** construction, so this approach must be replaced with new one.
  - I experimentally introduce `Base.assign` as replacement for `constructor`'s 2nd arg(was prop directly `Object.assign`-ed in constructor.

### CODE REVIEW

**Comment by t9md** [comment]
Will merge. I won't go back anyway.
I like JS now over CoffeeScript basically because  of prittier.
